corehq/apps/export/forms.py

```diff
@@ -501,23 +583,43 @@ def _get_datespan_filter(self):
             datespan.set_timezone(self.timezone)
             return ModifiedOnRangeFilter(gte=datespan.startdate, lt=datespan.enddate + timedelta(days=1))
 
-    def get_case_filter(self):
-        group = self._get_group()
-        if group:
-            user_ids = set(group.get_static_user_ids())
-            case_filter = [OR(
-                OwnerFilter(group._id),
-                OwnerFilter(user_ids),
-                LastModifiedByFilter(user_ids)
-            )]
+    def _get_location_filter(self, location_ids):
+        users = SQLLocation.users_at_locations_and_descendants(location_ids)
+        user_ids = [user['_id'] for user in users]
+        return OwnerFilter(user_ids)
+
+    def _get_group_independent_filters(self, mobile_user_and_group_slugs):
+        default_filters = [OwnerTypeFilter(self._get_es_user_types(mobile_user_and_group_slugs))]
+
+        location_ids = ExpandedMobileWorkerFilter.selected_location_ids(mobile_user_and_group_slugs)
+        if location_ids:
+            default_filters.append(self._get_location_filter(location_ids))
+
+        user_ids = ExpandedMobileWorkerFilter.selected_user_ids(mobile_user_and_group_slugs)
+        if user_ids:
+            default_filters.append(OwnerFilter(user_ids))
+
+        return default_filters
+
+    def get_case_filter(self, mobile_user_and_group_slugs=None):
+        print 'fetching filter for cases'
+        group_ids = self._get_group(mobile_user_and_group_slugs)
+
+        if group_ids:
+            groups_static_user_ids = Group.get_static_user_ids_for_groups(group_ids)
+            owner_filter_ids = group_ids + groups_static_user_ids
+            last_modified_filter_ids = groups_static_user_ids
         else:
-            case_sharing_groups = [g.get_id for g in
+            case_sharing_group_ids = [g.get_id for g in
                                    Group.get_case_sharing_groups(self.domain_object.name)]
-            case_filter = [OR(
-                OwnerTypeFilter(self._get_es_user_types()),
-                OwnerFilter(case_sharing_groups),
-                LastModifiedByFilter(case_sharing_groups)
-            )]
+            owner_filter_ids = case_sharing_group_ids
+            last_modified_filter_ids = case_sharing_group_ids
+
+        case_filter = [OR(
+            OwnerFilter(owner_filter_ids),
+            LastModifiedByFilter(last_modified_filter_ids),
+            *self._get_group_independent_filters(mobile_user_and_group_slugs)
+        )]
```

Comment:
A lot of this is similar to [`CaseListMixin.case_owners`](https://github.com/dimagi/commcare-hq/blob/c0dc05c0e5859dc7cd1152fc2927512c4c9fae49/corehq/apps/reports/standard/cases/basic.py#L114), which also uses the EMWF to filter cases.  I'm not sure if reusing code is doable (maybe!), but it could be useful at least as a reference.